src/apps/relay/ns_ioalib_engine_impl.c

```diff
@@ -3106,105 +3106,114 @@ int send_data_from_ioa_socket_nbh(ioa_socket_handle s, ioa_addr *dest_addr, ioa_
+    set_socket_ttl(s, ttl);
+    set_socket_tos(s, tos);
 
-          if (s->connected && s->bev) {
-            if ((s->st == TLS_SOCKET) || (s->st == TLS_SCTP_SOCKET)) {
+    if (s->connected && s->bev) {
+      if ((s->st == TLS_SOCKET) || (s->st == TLS_SCTP_SOCKET)) {
 #if TLS_SUPPORTED
-              SSL *ctx = bufferevent_openssl_get_ssl(s->bev);
-              if (!ctx || SSL_get_shutdown(ctx)) {
-                s->tobeclosed = 1;
-                ret = 0;
-              }
+        SSL *ctx = bufferevent_openssl_get_ssl(s->bev);
+        if (!ctx || SSL_get_shutdown(ctx)) {
+          s->tobeclosed = 1;
+          ret = 0;
+        }
 #endif
-            }
+      }
 
-            if (!(s->tobeclosed)) {
+      if (s->tobeclosed)
+        break;
 
-              ret = (int)ioa_network_buffer_get_size(nbh);
+      ret = (int)ioa_network_buffer_get_size(nbh);
 
-              if (!tcp_congestion_control || is_socket_writeable(s, (size_t)ret, __FUNCTION__, 2)) {
-                s->in_write = 1;
-                if (bufferevent_write(s->bev, ioa_network_buffer_data(nbh), ioa_network_buffer_get_size(nbh)) < 0) {
-                  ret = -1;
-                  perror("bufev send");
-                  log_socket_event(s, "socket write failed, to be closed", 1);
-                  s->tobeclosed = 1;
-                  s->broken = 1;
-                }
-                /*
-                bufferevent_flush(s->bev,
-                                                EV_READ|EV_WRITE,
-                                                BEV_FLUSH);
-                                                */
-                s->in_write = 0;
-              } else {
-                // drop the packet
-                ;
-              }
-            }
-          } else if (s->ssl) {
-            send_ssl_backlog_buffers(s);
-            ret = ssl_send(s, (char *)ioa_network_buffer_data(nbh), ioa_network_buffer_get_size(nbh),
-                           (s->e ? s->e->verbose : TURN_VERBOSE_NONE));
-            if (ret < 0)
-              s->tobeclosed = 1;
-            else if (ret == 0)
-              add_buffer_to_buffer_list(&(s->bufs), (char *)ioa_network_buffer_data(nbh),
-                                        ioa_network_buffer_get_size(nbh));
-          } else if (s->fd >= 0) {
-
-            if (s->connected && !(s->parent_s)) {
-              dest_addr = NULL; /* ignore dest_addr */
-            } else if (!dest_addr) {
-              dest_addr = &(s->remote_addr);
-            }
+      if (!tcp_congestion_control || is_socket_writeable(s, (size_t)ret, __FUNCTION__, 2)) {
+        s->in_write = 1;
+        if (bufferevent_write(s->bev, ioa_network_buffer_data(nbh), ioa_network_buffer_get_size(nbh)) < 0) {
+          ret = -1;
+          perror("bufev send");
+          log_socket_event(s, "socket write failed, to be closed", 1);
+          s->tobeclosed = 1;
+          s->broken = 1;
+        }
+        /*
+        bufferevent_flush(s->bev,
```

Comment:
Remove commented out code.

Reply:
I am keep old. if you want to remove it.please you are pr.